lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1589,6 +1595,9 @@ struct FIRRTLLowering : public FIRRTLVisitor<FIRRTLLowering, LogicalResult> {
   /// `getReadInOutOp(v)`.
   DenseMap<Value, Value> readInOutCreated;
 
+  /// Back-edges for each register.
+  DenseMap<Operation *, std::pair<Value, Backedge>> regMapping;
```

Comment:
Not sure if you saw this, but we have this utility on top of backedge builders: https://github.com/llvm/circt/blob/main/include/circt/Support/ValueMapper.h. Would that be useful here?

Reply:
I don't think it would be immediately useful here, but thanks for pointing me to this class.